tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -1848,6 +1838,105 @@ class ParameterizedOpConverterTestBase
   DataVec input_data_;
 };
 
+template <typename T>
+class OpConverter_UnaryTest : public ParameterizedOpConverterTestBase {
+ public:
+  template <typename S>
+  void RunTests(
+      const string& testName, std::vector<string>& ops_to_test,
+      const operationMap<S>& map,
+      std::map<std::string,
+               std::pair<std::function<NodeDef(DataType)>, T (*)(T)>>& op_map,
+      const std::vector<T> input_values, const std::string input_name = "input",
+      float max_abs_error = 0.0001, bool nan_sensitive = true,
+      bool checkOutput = false) {
+    // Prepare test parameters
+    auto p = TestParamBase{
+        {1, 1, 2, 3},  // input dims
+        {},            // input partial dims
+        {1, 1, 2, 3},  // expected output dims
+    };
+
+    for (auto& pair : map) {
+      ops_to_test.push_back(pair.first);
+    }
+
+    const auto& runtime_status = checkOutput ? Status::OK() : p.runtime_status;
+    for (const string& op_name : ops_to_test) {
+      SCOPED_TRACE(op_name);
+      if (!op_map.count(op_name)) {
+        FAIL() << testName << " op test map does not contain op " << op_name;
+      }
+
+      const DataType tf_type = get_tf_type();
+      const NodeDef& node_def = op_map[op_name].first(tf_type);
+      runExpectedToFailTest(node_def, input_name, input_values, op_name,
+                            checkOutput);
+
+      Status conv_status = Status::OK();
+      if (trt_mode_ == TrtTestMode::kImplicitBatch &&
+          (op_name == "Sign" || op_name == "Round" ||
+           op_name == "LogicalNot")) {
+        conv_status =
+            errors::Unimplemented("Unary op: '", op_name,
+                                  "' is not supported in implicit batch mode");
+      }
+
+      Reset();
+      // TODO(bixia): we assume this test is only instantiated for DT_FLOAT for
+      // now. Need to find a better way to express input and output types.
+      //
+      // TODO(tfeher): improve tests by defining an expected output data type
+      //  and check that. Currently only the shape and values of the output are
+      // checked.
+      const DataType input_tf_type = op_name == "Cast" ? DT_HALF : tf_type;
```

Comment:
Would this comment be proper for this line:
For CastOp, tf_type is the result type and we use DT_HALF as the input type.

Reply:
Currently cast only tests TD_HALF -> DT_FLOAT conversion. https://github.com/tensorflow/tensorflow/blob/24c3d0a72f534944de8505ccf13103534378469e/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc#L6906-L6907

A suggest to remove the comment, and add the following:
```suggestion
      const DataType input_tf_type = op_name == "Cast" ? DT_HALF : tf_type;
      const DataType output_tf_type = op_name == "Cast" ? DT_FLOAT : tf_type;
```
The `out_tf_type` variable shall be used while calling TestOpconverter.